packages/block-library/src/navigation/edit.js

```diff
@@ -126,56 +182,181 @@ function Navigation( {
 	}
 
 	function handleCreateFromExistingPages() {
-		updateNavItemBlocks( defaultPagesNavigationItems );
+		updateNavItemBlocks( navLinkBlocksFromPages );
 		selectBlock( clientId );
 	}
 
-	const hasPages = hasResolvedPages && pages && pages.length;
+	function handleCreateFromExistingMenu() {
+		updateNavItemBlocks( navLinkBlocksFromMenuItems );
+		selectBlock( clientId );
+	}
 
-	const blockInlineStyles = {
-		fontSize: fontSize.size ? fontSize.size + 'px' : undefined,
-	};
+	function handleCreate() {
+		const { key } = selectedDropDownOption;
+
+		// Explicity request to create empty.
+		if ( key === CREATE_EMPTY_OPTION_VALUE ) {
+			return handleCreateEmpty();
+		}
+
+		// Create from Pages.
+		if ( hasPages && key === CREATE_FROM_PAGES_OPTION_VALUE ) {
+			return handleCreateFromExistingPages();
+		}
+
+		// Create from WP Menu (if exists and not empty).
+		if (
+			hasMenus &&
+			selectedDropDownOption &&
+			navLinkBlocksFromMenuItems?.length
+		) {
+			return handleCreateFromExistingMenu();
+		}
+
+		// Default to empty menu
+		return handleCreateEmpty();
+	}
+
+	function buildPlaceHolderInstructionText() {
+		if ( hasMenus && hasPages ) {
```

Comment:
When loading a post that has an empty Navigation Block already inserted, I see a flicker of this text changing as pages/menus are loaded.

Reply:
I'll take a look.

Reply:
I think we should standardise on some simple unchanging text. Having to adapt to the option you can select is causing this jump. It would be best to avoid this.

Reply:
For those on slow connections the options in the dropdown will also be incorrect if they open the menu before loading has finished. Though I don't like it, I think a loading spinner may be the only option.

Reply:
> At the moment, placeholder content can only be shown when Menus and Pages have finished resolving, as that's needed to determine the content. A spinner should probably be used in the placeholder while that content is still loaded to avoid the content changing while being displayed (e.g. isRequestingMenus && isRequestingPages && ( <Spinner /> )).

> Though I don't like it, I think a loading spinner may be the only option.

I agree. I've updated to show a spinner within the placeholder if either pages or menus are still loading. Possibly annoying if you just want to create empty?